src/CMake/config/postinst.in

```diff
@@ -23,29 +23,40 @@
 # 3. When re-installing (say from 2.2.0 to 2.2.0) then prerm is NOT run but
 #    and postinst of 2.2.0 is run. The postinst is invoked with "2" argument.
 
+rmmodules()
+{
+    find /lib/modules -type f -name xocl.ko -delete
+    find /lib/modules -type f -name xclmgmt.ko -delete
+    find /lib/modules -type f -name xocl.ko.kz -delete
+    find /lib/modules -type f -name xclmgmt.ko.kz -delete
+    find /lib/modules -type f -name xocl.ko.xz -delete
+    find /lib/modules -type f -name xclmgmt.ko.xz -delete
+    depmod -A
+}
 
-# This snippet here to clean any previous XRT DKMS. This is always true for
-# RHEL/CentOS.
+lsb_release -si | grep -Eq "^Ubuntu|^Debian"
 
-if [ -n "`dkms status -m xrt`" ]; then
-    echo "Unloading old XRT Linux kernel modules"
-    modprobe -r xocl
-    modprobe -r xclmgmt
+if [ $? -eq 0 ] && [ "$1" = "configure" ]; then
+    # The older modules should already have been unregistered from dkms
+    # by the prerm
+    echo "Unloading old XRT Linux kernel modules on Ubuntu/Debian"
+    rmmod xocl
```

Comment:
It might be worth commenting why modprobe -r isn't used when all documentation says that rmmod should be avoided.